### DISCUSSION TITLE
Raman tensor

### DISCUSSION THREAD

**Discussion by eamouj78** [nguyen-group/QERaman#4] [Q&A] [answered]
Sorry to bother you again. I got the Raman tensor at the end in this form for the 6 modes in graphene:

iel =     1
       imode     1
           -0.37159591E-001  -0.52084047E-001   0.13240895E+000   0.19372395E+000  -0.17649985E-002  -0.85285974E-003
            0.13123341E+000   0.19603174E+000  -0.55708697E+000  -0.83045415E+000   0.19182040E-002   0.33954273E-002
           -0.16223602E-002  -0.10028858E-002   0.15134916E-002   0.42465079E-002  -0.96255900E-003   0.73297461E-005
       imode     2
           -0.26585471E-001  -0.47464614E-001   0.12016822E+000   0.19654698E+000   0.16474102E-002   0.79528175E-003
            0.12526254E+000   0.18780781E+000  -0.56933152E+000  -0.82210804E+000  -0.17942764E-002  -0.31708013E-002
            0.15135202E-002   0.93664677E-003  -0.14130627E-002  -0.39612914E-002  -0.12384872E-002   0.85248738E-005
       imode     3
           -0.24341952E-001  -0.46925004E-001   0.11871116E+000   0.19395248E+000   0.21419466E-002   0.14369943E-002
            0.11981237E+000   0.19211791E+000  -0.56292108E+000  -0.82651065E+000  -0.20289575E-002  -0.58334131E-002
            0.21459822E-002   0.10812116E-002  -0.23379332E-002  -0.40514232E-002  -0.12694978E-002   0.12332381E-004
       imode     4
            0.39665632E-001  -0.38162862E-001  -0.17566636E+000   0.15218742E+000   0.40775402E-003  -0.23708537E-002
           -0.17647463E+000   0.15372406E+000   0.75824689E+000  -0.65196752E+000  -0.70361579E-002   0.96142789E-002
           -0.12348243E-002  -0.62126469E-003   0.13461721E-002   0.23303317E-002  -0.73364526E-003   0.70712627E-005
       imode     5
           -0.63188046E-001  -0.10047349E+000  -0.14041033E+000   0.40326534E-001   0.10032682E-002   0.90981463E-002
           -0.15505757E+000   0.48039922E-001   0.99976989E+000  -0.21451481E-001   0.50860006E-001  -0.20388863E-001
            0.90796428E-002   0.56169331E-002  -0.85023172E-002  -0.23744854E-001  -0.55617229E-003   0.61687156E-005
       imode     6
            0.89964247E-001   0.19874744E-001  -0.25488075E+000  -0.33511382E-001  -0.73833215E-006   0.11328906E-005
           -0.27172025E+000  -0.24944500E-001   0.99666016E+000   0.81661001E-001  -0.51734809E-005  -0.81933241E-005
            0.71338285E-006   0.22891055E-006  -0.12473664E-004   0.23328142E-005  -0.21004546E-002   0.16022080E-004

Let us focus on, say imode6. 
1)Which represents the real part and the imaginary part of the tensor? How can we differentiate these?
2) How did you deduce that a and b (in the Raman tensors) are 0.1 and 1, respectively?

Thank you in advance

**Comment by nguyen-group** [comment, marked as answer]
Hi,

1) Let us take a set of values in imode 6 as (0.89964247E-001 0.19874744E-001), then 0.89964247E-001 is the real part, and 0.19874744E-001 is the imaginary part. 
2) In the example of graphene with 60x60x1, our calculated results as:
       imode     5
            0.11774040E-001   0.67078001E-002   **0.10368596E+000   0.99460619E+000**   0.27814410E-004  -0.35555809E-005
            **0.10368397E+000   0.99461029E+000**   0.15074375E-002  -0.73996073E-002  -0.11586300E-005  -0.14580835E-004
            0.26534192E-004   0.74163811E-005  -0.56446092E-005  -0.43630915E-007   0.48308319E-003  -0.38809653E-005
       imode     6
            **0.10460818E+000   0.99386902E+000**   0.35508057E-002  -0.74211520E-002  -0.25745551E-005  -0.10840885E-005
            0.35575602E-002  -0.74204142E-002  **-0.11627329E+000  -0.99321726E+000**   0.13699722E-004  -0.91606077E-005
           -0.55019198E-005  -0.87527308E-005   0.28276571E-004  -0.18669124E-005  -0.28221599E-003   0.22620888E-005

You can see that a+ib = 0.10460818   + i0.99386902, so that is why a = 0.1 and b =0.99.
I note that a and b are very sensitive to the selected k-points. There is no rule for the convergence test of k-points for a and b. However, if the (a^2 + b^2) values of R_11 ~ R_22 and much higher than other Raman tensor components, as shown in bold text in imode 6, we can say it is a good choice. 

In your case, you might use the different k-points or parameters compared with the example. We appreciate if you could do more tests with other k-points to check the value of a and b.

**Reply by eamouj78** [reply]
Once more, thank you very much for your time and patience. 

The results I showed were a quick test to see if I can manage to obtain the Raman spectra using the raman.x executable. I will do more rigorous tests to investigate the convergence of "a" and "b" with denser k-points.

**Comment by eamouj78** [comment]
Dear all, 

I have been performing some tests on the graphene Raman spectra. My results converged for 96x96x1. However I am getting completely different results in contradiction with the results reported in the example. The s+s+ values are very high, whereas they should be very low:

Here is a sample for a 72x72x1 grid :

**My results:**

#Raman spectra
#Laser energy   2.33 eV
#Raman shift (cm-1), Raman Spectra (XX,YY,ZZ,s+s+,s-s-,XY,YX,s+s-,s-s+)
  1500.0000  0.28451797E+002  0.12499374E+004  0.10303849E+000  **0.33009247E+003**  0.31607869E+003  0.44149220E+002  0.42980392E+002  0.35006759E+003  0.36928010E+003
  1500.1821  0.28623881E+002  0.12528658E+004  0.10358907E+000  0.33088534E+003  0.31677346E+003  0.44287541E+002  0.43115356E+002  0.35137037E+003  0.36986340E+003
  1500.3643  0.28797674E+002  0.12558251E+004  0.10414516E+000  0.33168656E+003  0.31747562E+003  0.44427288E+002  0.43251711E+002  0.35268642E+003  0.37045316E+003
  1500.5464  0.28973199E+002  0.12588157E+004  0.10470682E+000  0.33249626E+003  0.31818526E+003  0.44568479E+002  0.43389476E+002  0.35401592E+003  0.37104946E+003
  1500.7286  0.29150478E+002  0.12618382E+004  0.10527413E+000  0.33331454E+003  0.31890250E+003  0.44711134E+002  0.43528669E+002  0.35535903E+003  0.37165239E+003
  1500.9107  0.29329536E+002  0.12648928E+004  0.10584716E+000  0.33414151E+003  0.31962742E+003  0.44855273E+002  0.43669309E+002  0.35671594E+003  0.37226203E+003
  1501.0929  0.29510396E+002  0.12679800E+004  0.10642599E+000  0.33497730E+003  0.32036014E+003  0.45000915E+002  0.43811415E+002  0.35808684E+003  0.37287849E+003

**Results reported in code:**

1500.0000  0.64057143E+003  0.63952548E+003  0.10761745E-003  **0.28166636E-001**  0.28180382E-001  0.64062004E+003  0.64062193E+003  0.12806222E+004  0.12806603E+004
  1500.2506  0.64880016E+003  0.64774078E+003  0.10899987E-003  0.28528442E-001  0.28542365E-001  0.64884957E+003  0.64885148E+003  0.12970732E+004  0.12971118E+004
  1500.5013  0.65718838E+003  0.65611530E+003  0.11040909E-003  0.28897261E-001  0.28911363E-001  0.65723861E+003  0.65724055E+003  0.13138430E+004  0.13138821E+004
  1500.7519  0.66574024E+003  0.66465320E+003  0.11184580E-003  0.29273275E-001  0.29287560E-001  0.66579131E+003  0.66579328E+003  0.13309399E+004  0.13309795E+004
  1501.0025  0.67446002E+003  0.67335873E+003  0.11331072E-003  0.29656672E-001  0.29671144E-001  0.67451195E+003  0.67451394E+003  0.13483726E+004  0.13484127E+004
  1501.2531  0.68335214E+003  0.68223633E+003  0.11480459E-003  0.30047647E-001  0.30062309E-001  0.68340495E+003  0.68340696E+003  0.13661498E+004  0.13661905E+004
  1501.5038  0.69242116E+003  0.69129055E+003  0.11632819E-003  0.30446401E-001  0.30461257E-001  0.69247488E+003  0.69247692E+003  0.13842807E+004  0.13843219E+004
  1501.7544  0.70167182E+003  0.70052610E+003  0.11788230E-003  0.30853140E-001  0.30868194E-001  0.70172646E+003  0.70172853E+003  0.14027747E+004  0.14028165E+004
  1502.0050  0.71110899E+003  0.70994786E+003  0.11946774E-003  0.31268081E-001  0.31283337E-001  0.71116458E+003  0.71116667E+003  0.14216416E+004  0.14216839E+004

What could have possibly gone wrong

**Comment by nguyen-group** [comment]
Hi,

I don't have any idea to see the output. Could you give me the input file?

**Reply by eamouj78** [reply]
My fault. I should have sent the input files. Since I could not attaching. I will copy-paste

Regards

**scf.in:** 

&control
 prefix='graphene'
    calculation='scf',
     tstress=.true.,
    tprnfor=.true.,
    pseudo_dir = '~/PP/'
    wf_collect=.true.
    outdir='/scratch/90022a/eamouj78/graphene'
    wfcdir='/scratch/90022a/eamouj78/graphene'
   etot_conv_thr=1.0D-6
    forc_conv_thr=1.0D-5
/
&system
    ibrav=  4, celldm(1) =4.6614811157162, celldm(3)=6.5, nat=2, ntyp= 1, ecutwfc = 70, ecutrho=700, degauss=0.007, nbnd=10, occupations='smearing', smearing='gaussian', assume_isolated='2D'

/
 &electrons
    conv_thr=1.D-8,
    electron_maxstep=300
    mixing_beta=0.1D0,
    mixing_mode='plain'
         /
&ions
 ion_dynamics ='bfgs'
/
 &cell
 cell_dynamics='bfgs'
 press=0.0
 cell_dofree='fixc'
/
ATOMIC_SPECIES
 C  12.0107  C.pbe-paw.upf

ATOMIC_POSITIONS crystal
 C        0.000000000  -0.000000000   0.000000000 
 C        0.333333000  -0.333330000  0.000000000 

K_POINTS automatic
72 72 1 0 0 0

**bands_mat file:**
 &BANDS
 prefix = 'graphene'
 outdir='/scratch/90022a/eamouj78/graphene'
 filband = 'graphene.bands'
 lp = .true.
 filp = 'graphene.dvec'
 /

**ph_mat file:**

phonons
&inputph
prefix = 'graphene'
recover=.true.
outdir ='/scratch/90022a/eamouj78/graphene' 
tr2_ph = 1.0d-18
fildyn = 'graphene.dyn'
fildvscf = 'dvscf'
electron_phonon = 'epc'
/
0.0 0.0 0.0

**raman.in file:**

&inputraman
 prefix = 'graphene'
 outdir='~/graphene/RAMAN/72x72x1/'
 fil_dvec = 'graphene.dvec'
 fil_elph = 'graphene.elph'
 sorb = .false.
 circular_pol =.true.
 nonpol = .false.
 plot_matele_opt =.false.
 plot_matele_elph = .false.
 plot_raman_k = .false.
 gamma = 0.05
 gamma_raman = 0.0002
 rs_start = 1500.0
 rs_end = 1600.0
 nrs = 550
elaser1 = 2.33
 /

**Reply by nguyen-group** [reply]
The inputs seem correct.
Even our code can run with PAW, but I am not sure whether the electron-phonon from QE can work properly with PAW or not.
So I suggest changing C.pbe-paw.upf to another NC or US for testing. In my example, I used C.pz-hgh.UPF.

**Reply by eamouj78** [reply]
Thank you for your reply. 

Since the code did not complain, I thought it/QE works well with PAW. I will test with a NC PP and let you know.

Regards

**Comment by eamouj78** [comment]
Hi again,

I found what the problem is. The flags noinv and nosym must be set to true; otherwise the results are different.  Something goes terribly wrong when the symmetries are switched on. I do not understand why. Is setting these flags to .true. a necessity for things to work?
Using the same C.pz-hgh.UPF PP:

**Results with symmetry (nosym and noinv set to .false.)**
#Raman spectra
#Laser energy   2.33 eV
#Raman shift (cm-1), Raman Spectra (XX,YY,ZZ,s+s+,s-s-,XY,YX,s+s-,s-s+)
  1500.0000  0.10066241E+002  0.60146413E+003  **0.12629235E+000  0.14578244E+003**  0.14218476E+003  0.22148823E+002  0.27692059E+002  0.16844818E+003  0.20495587E+003
  1500.1821  0.10146264E+002  0.60447220E+003  0.12725882E+000  0.14648761E+003  0.14285723E+003  0.22276515E+002  0.27865072E+002  0.16932385E+003  0.20609136E+003
  
**Results with NO symmetry (nosym and noinv set to true.)**
1500.0000  0.38484584E+003  0.38188234E+003  **0.12586219E-003  0.12641939E-001**  0.12509056E-001  0.38424858E+003  0.38426532E+003  0.76733604E+003  0.76788090E+003
  1500.1821  0.38794558E+003  0.38495822E+003  0.12687503E-003  0.12743474E-001  0.12609523E-001  0.38734194E+003  0.38735882E+003  0.77351499E+003  0.77406423E+003


Regards

**Comment by eamouj78** [comment]
Hi again,

I just tried it with another PP, and I am getting the same results you reported.

The culprit (at least for the case of graphene) is indeed the nosym, noinv flags.

**Reply by nguyen-group** [reply]
Hi, I see. Sorry that I did not check scf.in carefully. No nosym and noinv must to set in the scf.in. The reason was explained in the paper. By using this option, both el-photon and el-phonon will be calculated with the same k- and q-mesh.

**Reply by eamouj78** [reply]
Yes, sorry, I have not seen that we need to set these flags to .true.

A final question: you mention that the flag "gamma" needs to be set according to the experiment.
Let us say there is no experimental reference, and the structure is still new theoretically; how can we choose values of "gamma" and "gamma_raman"? Do these need to be converged, or are the ones included in the examples OK?

Regards

**Reply by nguyen-group** [reply]
That is a good question. I don't think the convergence can work with gamma and gamma_raman. For a new material, a plot of Raman as a function of gamma might be helpful. You can see in Fig. of this paper: https://journals.aps.org/prb/supplemental/10.1103/PhysRevB.97.115407/tatsumi17_hrraman-SI.pdf

**Reply by eamouj78** [reply]
Thank you very much for your help and patience.
I will have a close look at that paper.

Regards
